app/models/pusher.rb

```diff
@@ -45,7 +60,15 @@ def save
   end
 
   def pull_spec
-    @spec = Gem::Package.new(body).spec
+    # Verify that the gem signatures match the certificate chain (if present)
+    policy = Gem::Security::LowSecurity.dup
+    # Silence warnings from the verification
+    null_in = File.open(IO::NULL, "r")
```

Comment:
```suggestion
    stream = StringIO.new
```

An later...

```ruby
policy.ui = Gem::StreamUI.new(stream, stream, stream, false)
```

With this, the whole system will not have the problem of dangling file descriptors still opened on the host because has not been closed properly. As long as I remember `Grem::StreamUI` will not try to close the IO passed on their parameters

Reply:
Thanks for the suggestion, I pushed a fix to the branch